File: cadwiki-nuget/cadwiki.AutoCAD2021.Base.Utilities/Shared/Polylines.vb
Option Strict On
Option Infer Off
Option Explicit On

Imports Autodesk.AutoCAD.BoundaryRepresentation
Imports Autodesk.AutoCAD.DatabaseServices
Imports Autodesk.AutoCAD.Geometry

Public Class Polylines
    Public Shared Function IsPointInside(ByVal pline As Polyline, ByVal point As Point3d) As Boolean
        Dim dbCollection As DBObjectCollection = New DBObjectCollection()
        dbCollection.Add(pline)

        Dim regionCollection As DBObjectCollection = Region.CreateFromCurves(dbCollection)
        Dim acRegion As Region = CType(regionCollection(0), Region)

        Dim pointCont As PointContainment = New PointContainment

        Dim brep As Brep = New Brep(acRegion)

        brep.GetPointContainment(point, pointCont)
        If Not pointCont = PointContainment.Outside Then
            Return True
        End If
        Return False
    End Function
End Class

File: cadwiki-nuget/cadwiki.AC/Utilities/Polylines.cs

using Autodesk.AutoCAD.BoundaryRepresentation;
using Autodesk.AutoCAD.DatabaseServices;
using Autodesk.AutoCAD.Geometry;

namespace cadwiki.AC
{

    public class Polylines
    {
        public static bool IsPointInside(Polyline pline, Point3d point)
        {
            var dbCollection = new DBObjectCollection();
            dbCollection.Add(pline);

            var regionCollection = Region.CreateFromCurves(dbCollection);
            Region acRegion = (Region)regionCollection[0];

            var pointCont = new PointContainment();

            var brep = new Brep(acRegion);

            brep.GetPointContainment(point, out pointCont);
            if (!(pointCont == PointContainment.Outside))
            {
                return true;
            }
            return false;
        }
    }
}